electrum/plugins/bitbox02/bitbox02.py

```diff
@@ -0,0 +1,628 @@
+                    bitbox02_config["remote_static_noise_keys"].append(pubkey.hex())
+                else:
+                    bitbox02_config["remote_static_noise_keys"] = [pubkey.hex()]
+                self.config.set_key("bitbox02", bitbox02_config)
+
+        def get_noise_privkey() -> Optional[bytes]:
+            bitbox02_config = self.config.get("bitbox02")
+            privkey = bitbox02_config.get("noise_privkey")
+            if privkey is not None:
+                return bytes.fromhex(privkey)
+            return None
+
+        def set_noise_privkey(privkey: bytes) -> None:
+            bitbox02_config = self.config.get("bitbox02")
+            bitbox02_config["noise_privkey"] = privkey.hex()
+            self.config.set_key("bitbox02", bitbox02_config)
+
+        def attestation_warning() -> None:
+            self.handler.attestation_failed_warning(
+                "The BitBox02 attestation failed.\nTry reconnecting the BitBox02.\nWarning: The device might not be genuine, if the\n problem persists please contact Shift support."
+            )
+
+        class NoiseConfig(bitbox_api_protocol.BitBoxNoiseConfig):
+            """NoiseConfig extends BitBoxNoiseConfig"""
+
+            def show_pairing(self, code: str) -> bool:
+                choice = [code]
+                try:
+                    reply = pairing_step(code)
+                except:
+                    # Close the hid device on exception
+                    hid_device.close()
+                    raise
+                return True
+
+            def attestation_check(self, result: bool) -> None:
+                if not result:
+                    attestation_warning()
+
+            def contains_device_static_pubkey(self, pubkey: bytes) -> bool:
+                return exists_remote_static_pubkey(pubkey)
+
+            def add_device_static_pubkey(self, pubkey: bytes) -> None:
+                return set_remote_static_pubkey(pubkey)
+
+            def get_app_static_privkey(self) -> Optional[bytes]:
+                return get_noise_privkey()
+
+            def set_app_static_privkey(self, privkey: bytes) -> None:
+                return set_noise_privkey(privkey)
+
+        if self.bitbox02_device is None:
+            hid_device = hid.device()
+            hid_device.open_path(self.bitbox_hid_info["path"])
+
+            self.bitbox02_device = bitbox02.BitBox02(
+                transport=u2fhid.U2FHid(hid_device),
+                device_info=self.bitbox_hid_info,
+                noise_config=NoiseConfig(),
+            )
+
+        if not self.bitbox02_device.device_info()["initialized"]:
+            raise BaseException(
+                "Please initialize the BitBox02 using the BitBox app first before using the BitBox02 in electrum"
+            )
+
+    def check_device_firmware_version(self) -> bool:
+        if self.bitbox02_device is None:
+            raise BaseException(
```

Comment:
Try never to raise BaseException. Just raise Exception instead.

Reply:
Done.